pkg/ottl/ottlfuncs/func_replace_pattern.go

```diff
@@ -36,6 +37,47 @@ func createReplacePatternFunction[K any](_ ottl.FunctionContext, oArgs ottl.Argu
 	return replacePattern(args.Target, args.RegexPattern, args.Replacement, args.Function)
 }
 
+// function to check if a string is a group number string like "$1", "$2", ...
+func IsGroupNumString(s string) bool {
+	match, _ := regexp.MatchString(`^\$\d+$`, s)
+	return match
+}
+
+func applyReplaceFunction[K any](ctx context.Context, tCtx K, compiledPattern *regexp.Regexp, fn ottl.Optional[ottl.FunctionGetter[K]], originalValStr string, replacementVal string) (string, error) {
```

Comment:
What's the difference between this function and the replace_all variant? Would it be feasible to combine the two?

Reply:
Its the same function, I changed the code to use one function.